Fix warn_unused_result in hyperbolic_upwind.cpp

There is a warning in the compile for hyperbolic1D_upwind.cpp:
```
mole/examples/cpp/hyperbolic1D_upwind.cpp:127:11: warning: ignoring return value of ‘int system(const char*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  127 |     system("gnuplot -persistent gp_script");
```
which should be rectified.

--- a/examples/cpp/hyperbolic1D_upwind.cpp
+++ b/examples/cpp/hyperbolic1D_upwind.cpp
@@ -124,6 +124,13 @@ int main()
     scriptFile.close();
 
     // Run GNUplot with the script
-    system("gnuplot -persistent gp_script");
+    int status = std::system("gnuplot -persistent gp_script");
+
+    if ( status != 0 ) {
+        std::cerr << "Error executing GNUplot.\n";
+        return EXIT_FAILURE;
+    }
+ 
+
     return EXIT_SUCCESS;
 }

